app/src/androidTest/java/com/github/h3lp3rs/h3lp/forum/ForumPostsActivityTest.kt

```diff
@@ -0,0 +1,65 @@
+package com.github.h3lp3rs.h3lp.forum
+
+import android.content.Intent
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.intent.Intents.*
+import androidx.test.espresso.intent.matcher.IntentMatchers
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.h3lp3rs.h3lp.R
+import com.github.h3lp3rs.h3lp.USER_TEST_ID
+import com.github.h3lp3rs.h3lp.signin.SignInActivity
+import org.hamcrest.Matchers
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.mockito.kotlin.any
+
+@RunWith(AndroidJUnit4::class)
+class ForumPostsActivityTest {
+    private val forumPosts: MutableMap<String, List<String>> = mutableMapOf()
+
+    @Before
+    fun setup() {
+        SignInActivity.globalContext = ApplicationProvider.getApplicationContext()
+        SignInActivity.userUid = USER_TEST_ID
+        val intent = Intent(
+            ApplicationProvider.getApplicationContext(),
+            ForumPostsActivity::class.java
+        ).apply {
+            putExtra(EXTRA_FORUM_CATEGORY, CATEGORY_TEST_STRING)
+        }
+
+        ActivityScenario.launch<ForumPostsActivity>(intent)
```

Comment:
for consistency we could launch right before init(), or is there a specific reason you do it before mocking?

Reply:
Same, this ordering is weird and non intuitive Mocking should come before as well